### PR TITLE
Implemented page_size parameter to allow users to set page size

### PR DESCRIPTION
**Pull request by noisecapella**
Required for #613 

Allows us to optionally set a `page_size` parameter which lets us gulp down data more efficiently

### CODE REVIEW

File: rest/pagination.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from rest_framework.pagination import PageNumberPagination
+
+
+class LorePagination(PageNumberPagination):
+    """
+    Pagination class for Lore's REST API.
+    """
+    page_size = 20
+    page_size_query_param = 'page_size'
+    max_page_size = 1000
```
Comment (giocalitri) [inline rest/pagination.py:16]: where are these numbers coming from?

Comment (noisecapella) [inline rest/pagination.py:16]: 20 is the default page size, 1000 is just some number I considered to be big enough for one request, a guess

Comment (giocalitri) [inline rest/pagination.py:16]: is 20 the same number in the configuration? if so you should take it from there

File: rest/tests/test_misc.py [hunk 2]
```diff
@@ -48,3 +53,41 @@ def test_root(self):
...
+        """
+        Test page_size query parameter.
+        """
+        self.import_course_tarball(self.repo)
+
+        num_resources = LearningResource.objects.filter(
+            course__repository__id=self.repo.id
+        ).count()
+
+        def assert_page_size(page_size):
```
Comment (giocalitri) [inline rest/tests/test_misc.py]: nit: I would move this function at the beginning, after the doctoring of the parent function

File: rest/pagination.py (hunk 1 above)
Comment (noisecapella) [inline rest/pagination.py:16]: Sorry, it's not the default, it was a value we added. It's removed from lore/settings.py above so this is the only place it exists now.

Comment (giocalitri) [inline rest/pagination.py:16]: sorry, now I understand the first file (that I completely skipped)

File: rest/tests/test_misc.py [hunk 3]
```diff
@@ -48,3 +53,41 @@ def test_root(self):
...
+                )
+            )
+            self.assertEqual(resp.status_code, HTTP_200_OK)
+            resources = as_json(resp)
+            self.assertTrue(len(resources['results']) <= page_size)
+            self.assertEqual(resources['count'], num_resources)
+
+        assert_page_size(2)
+        assert_page_size(10)
+        assert_page_size(None)
```
Comment (giocalitri) [inline rest/tests/test_misc.py]: what happens with `assert_page_size(LorePagination.max_page_size +1)` ?

Comment (giocalitri) [inline rest/tests/test_misc.py]: what about invalid page numbers?

File: ui/tests/test_repository_views.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        Test that page size context is set properly.
+        """
+        resp = self.client.get(
+            "/repositories/{slug}/".format(slug=self.repo.slug))
+        self.assertEqual(json.loads(resp.context['page_size_json']), 20)
+
+        # Invalid params are ignored.
+        resp = self.client.get(
+            "/repositories/{slug}/?page_size=xyz".format(slug=self.repo.slug))
+        self.assertEqual(json.loads(resp.context['page_size_json']), 20)
```
Comment (giocalitri) [inline ui/tests/test_repository_views.py]: ok, fount the answer to my previous question (invalid page number)

File: ui/tests/test_repository_views.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+        resp = self.client.get(
+            "/repositories/{slug}/?page_size=300".format(slug=self.repo.slug))
+        self.assertEqual(json.loads(resp.context['page_size_json']), 300)
```
Comment (giocalitri) [inline ui/tests/test_repository_views.py]: I guess you need an assert for `page_size` > max allowed

File: rest/tests/test_misc.py (hunk 3 above)
Comment (noisecapella) [inline rest/tests/test_misc.py]: I'll add this, it works fine locally

File: rest/tests/test_misc.py (hunk 2 above)
Comment (noisecapella) [inline rest/tests/test_misc.py]: Will do

File: ui/tests/test_repository_views.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            'manage_repo_users',
+            'add_edit_metadata',
+            'manage_taxonomy',
+            'view_repo',
+            'import_course'
+        ]
+        self.assertEqual(
+            sorted(resp.context["perms_on_cur_repo"]),
+            sorted(expected_perms)
+        )
```
Comment (giocalitri) [inline ui/tests/test_repository_views.py:91]: why do you need this test?

Comment (giocalitri) [inline ui/tests/test_repository_views.py:91]: sorry, now I understand, it's for the context

**Comment by giocalitri** [comment]
This is good to go as soon as you squash and tests pass

**Pull request merged**